00 - Contribute to the Obsidian Hub/03 Contributor Notes/03.02 Design Decisions/hub.yaml proposal for themes.md

```diff
@@ -0,0 +1,303 @@
+
+## Current workflow
+
+As you may know, some of the hub content is populated via scripts. For themes, there are two sources of information we use:
+- What you submitted to the `obsidian-releases` repo, e.g. 
+  ```json
+      {
+        "name": "Shimmering Focus",
+        "author": "pseudometa",
+        "repo": "chrisgrieser/shimmering-focus",
+        "screenshot": "docs/images/Promo%20Screenshot/promo-screenshot.png",
+        "modes": ["dark", "light"],
+        "branch": "main"
+    },
+   ```
+- Your `obsidian.css` file to determine if you use the Style settings plugin and to see if you've listed any [compatible plugins](https://publish.obsidian.md/hub/04+-+Guides%2C+Workflows%2C+%26+Courses/Guides/Adding+plugin+compatibility+for+themes+to+the+Obsidian+Hub).
+
+This is covers the very basic information for most themes, but
+- is not very user-friendly when trying to find themes that have X (which themes support asides? highlights?) and users have to try out the themes to find out
+- limits you to a single screenshot
+- requires manual edits on your side if you want to add text or other features not mentioned in style settings or list sponsorship information
+	- content you add to the hub is  outside to your theme repo and will easily get out of date
+	- adding or updating any content you do add to the hub requires you to go through the fork and PR workflow, which adds friction and makes it less likely that you do either
+- requires manual work in the hub  repo when there are manual changes to make sure the scripts haven't removed manually added content and handling other edge cases (at least for now)
+
+## Proposal
+
+Our proposal is to add an optional `hub.yaml` file at the root of theme repos to add some of this information. Note that all entries are optional, so maintenance and effort can be managed by each theme designer based on the their theme's complexity and requirements. We'll cover some of the basics with examples below, but you can find the full file with all the options further down in the [[#Template]] section.
+
+### Link to the theme's documentation
+
+The contents of the `hub.yaml` for a theme that **only** wants to add a link to the theme's documentation in its hub note would be as simple as:
+
+```yaml
+docs: https://minimal.guide/ # Link to the theme documentation
+```
+
+### Hub note contents
+
+Similarly, if you wanted to have some control on the contents of your theme note, you can use the `hub_content` key to list which markdown documents to "import" to the hub. Generally, to avoid having to maintain two sets of docs (one for your theme and one for the hub), you could reuse a markdown file with an overview that has links to your docs.
+
+For example, with the contents of `hub.yaml` as show below:
+```yaml
+hub_content: # List of markdown documents to be included as is to the hub relative to the root of the repo.
+	- docs/theme_overview.md
+```
+
+And the contents of `theme_overview.md`:
+```markdown
```

Comment:
Any thoughts about relative links in the markdown? I suspect we will soon end up having to parse the imported markdown files and add URLs where the markdown had internal links within the theme's repo.

Reply:
I'd start with "as is", no parsing or adding any other links. If there are links, we should recommend that they are always external (better for reuse) or use `[[wikilinks]]` of notes in the hub (if they are making such file specifically for us).

Reply:
Sounds good. If that's not already in the doc, might it be possible to include that?